src/main/java/org/springframework/data/elasticsearch/core/query/AbstractQuery.java

```diff
@@ -160,4 +164,13 @@ public IndicesOptions getIndicesOptions() {
 	public void setIndicesOptions(IndicesOptions indicesOptions) {
 		this.indicesOptions = indicesOptions;
 	}
+
+	@Override
+	public boolean getTrackScores() {
+		return trackScores;
+	}
+
+	public void setTrackScores(boolean trackScores) {
```

Comment:
At a quick glance, this setter doesn't seem to be used anywhere, right?

Reply:
Oh, it is. From `NativeSearchQueryBuilder`. However, it should be in package scope then to limit the API surface.

Reply:
This setter needs to be public for cases where `NativeSearchQueryBuilder` is not used.

Reply:
I am probably too unfamiliar with the ES specifics here. I am merely looking at this from an "How much API do we expose?" point of view and API that assumes or exposes mutability is usually candidate for extra scrutiny.

Reply:
There are cases in which the query is created using the constructors of `NativeSearchQuery` and then additional setters are called. If we want to enforce the use of the `NativeSearchQueryBuilder`, all constructors and setters should be package scoped. But we should not do that, as it would break existing code. So in my opinion, this setter should be public analog to the other setters. In our internal projects, we implemented a less complex search query builder implementation which needs to have access to the `setTrackScores` method.

Reply:
Understood, thanks. Looks like Sascha opened up the visibility to `public` again.

Reply:
Yeah we discussed it before.